flight/PiOS/Common/pios_dsm.c

```diff
@@ -316,9 +316,40 @@ int32_t PIOS_DSM_Init(uintptr_t *dsm_id,
 	/* Bind the configuration to the device instance */
 	dsm_dev->cfg = cfg;
 
+	uint8_t num_pulses = 0;
+
+	// check user settings to determine which resolution mode to use
+	// if invalid mode selected, bail
+	if (mode > HWSHARED_DSMXMODE_BIND10PULSES)
```

Comment:
@mlyle For 2), I'm mainly thinking about these enums in the context of generated UAVOs, in which case we have complete control over the starting value and increment of each element and can ensure they're generated in such a way that a range check will always work. I agree with 4).